pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -184,11 +224,45 @@ func (c *Reconciler) updateStatus(taskrun *v1alpha1.TaskRun) (*v1alpha1.TaskRun,
 	if err != nil {
 		return nil, err
 	}
-	if !reflect.DeepEqual(newtaskrun.Status, taskrun.Status) {
+	if !reflect.DeepEqual(taskrun.Status, taskrun.Status) {
 		newtaskrun.Status = taskrun.Status
 		// TODO: for CRD there's no updatestatus, so use normal update
 		return c.PipelineClientSet.PipelineV1alpha1().TaskRuns(taskrun.Namespace).Update(newtaskrun)
 		//	return configClient.UpdateStatus(newtaskrun)
 	}
 	return newtaskrun, nil
 }
+
+func (c *Reconciler) getBuild(namespace, name string) (*buildv1alpha1.Build, error) {
+	return c.buildLister.Builds(namespace).Get(name)
+}
```

Comment:
- [ ] i dont think we need these wrapper methods, this might have been based on a previous iteration of the pipelinerun controller (for mocking), check out the latest greatness https://github.com/knative/build-pipeline/blob/master/pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go#L69